packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -183,25 +218,68 @@ export default function VisualEditor( { styles } ) {
 
 	const blockSelectionClearerRef = useBlockSelectionClearer();
 
-	const layout = useMemo( () => {
+	// fallbackLayout is used if there is no Post Content,
+	// and for Post Title.
+	const fallbackLayout = useMemo( () => {
 		if ( isTemplateMode ) {
 			return { type: 'default' };
 		}
 
 		if ( themeSupportsLayout ) {
 			// We need to ensure support for wide and full alignments,
 			// so we add the constrained type.
-			return { ...defaultLayout, type: 'constrained' };
+			return { ...globalLayoutSettings, type: 'constrained' };
 		}
 		// Set default layout for classic themes so all alignments are supported.
 		return { type: 'default' };
-	}, [ isTemplateMode, themeSupportsLayout, defaultLayout ] );
+	}, [ isTemplateMode, themeSupportsLayout, globalLayoutSettings ] );
+
+	const postContentBlock = useMemo( () => {
+		// When in template editing mode, we can access the blocks directly.
+		if ( editedPostTemplate?.blocks ) {
+			return findPostContent( editedPostTemplate?.blocks );
+		}
+		// If there are no blocks, we have to parse the content string.
+		// Best double-check it's a string otherwise the parse function gets unhappy.
+		const parseableContent =
+			typeof editedPostTemplate?.content === 'string'
+				? editedPostTemplate?.content
+				: '';
+
+		return findPostContent( parse( parseableContent ) ) || {};
+	}, [ editedPostTemplate?.content, editedPostTemplate?.blocks ] );
+
+	const postContentLayoutClasses = useLayoutClasses( postContentBlock );
+
+	const blockListLayoutClass = classnames(
+		{
+			'is-layout-flow': ! themeSupportsLayout,
+		},
+		themeSupportsLayout && postContentLayoutClasses
+	);
 
-	const blockListLayoutClass = classnames( {
-		'is-layout-constrained': themeSupportsLayout,
-		'is-layout-flow': ! themeSupportsLayout,
-		'has-global-padding': useRootPaddingAwareAlignments,
-	} );
+	const postContentLayoutStyles = useLayoutStyles(
+		postContentBlock,
+		'.block-editor-block-list__layout.is-root-container'
+	);
```

Comment:
I think there appears to be a difference in the rules that are being output here versus trunk 🤔 I noticed that there's an issue with the middle content justification with the Cover block set to wide alignment in TwentyTwentyTwo:

| Trunk | This PR |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/192202854-7f45060a-745c-48a8-9434-c3ad6dccd1d5.png) | ![image](https://user-images.githubusercontent.com/14988353/192203014-68708d70-0111-4572-a7b0-e0ce7219a989.png) |

This appears to be the rule that isn't present / winning out:

<img width="425" alt="image" src="https://user-images.githubusercontent.com/14988353/192203888-6d2f297d-a645-4b28-8581-d6565c6d317c.png">

Reply:
I wonder if the issue is that `useLayoutStyles` generates the layout styles for the individual block, but not the base layout styles that need to be attached to this more specific selector in order to win out?

Reply:
> I'm reluctant to either increase specificity of the default layout styles or add wp-container rules for the default justification state

Agreed, it'd be better not to have to add container rules, or explicit rules for the default justification state 🤔

> This should probably be fixed in TT2 instead

It's a tricky one... because TT2 has been used as a base for many other themes, it might not just TT2 out in the wild that has this issue. (E.g. the rule appears to be part of BlockBase https://github.com/Automattic/themes/blob/23b9090090279c95788357bb634fd2e57c3973db/blockbase/sass/base/_alignment.scss#L58 which many themes have been built on top of, too).

Just trying to think of other options... would it work to update the change to this line in this PR https://github.com/WordPress/gutenberg/blob/67c829dc9d4a265c22961623c02c42185321c2b2/packages/edit-post/src/components/visual-editor/index.js#L340 to include `.block-editor-block-list__layout.is-root-container` as well as on trunk? Or did we we need to remove it for the `postContentLayoutStyles` to work? Since they already have the selector attached, I was wondering if they can coexist. From quickly adding it back in in my local environment, it appears to get all three content justifications working in TT2 for me, but it's highly likely I'm missing some context 😅

Reply:
> Just trying to think of other options... would it work to update the change to this line in this PR to include `.block-editor-block-list__layout.is-root-container` as well (as on trunk)? Or did we we need to remove it for the `postContentLayoutStyles` to work?

I did remove it initially so that custom content widths would work, but because we since increased the specificity of the default rules it seems to be working ok. I just tried pushing that change now, hopefully it doesn't break anything else 😅  

> It's a tricky one... because TT2 has been used as a base for many other themes, it might not just TT2 out in the wild that has this issue.

I'm not sure how much of a concern that should be to us, because the temporary status of these rules is explicitly called out in the TT2 [stylesheet](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-content/themes/twentytwentytwo/style.css#L87).